Wikipedia/Code/DiffHeaderView.swift

```diff
@@ -0,0 +1,89 @@
+import UIKit
+import CocoaLumberjackSwift
+
+final class DiffHeaderView: SetupView {
+
+    var viewModel: DiffHeaderViewModel?
+
+    lazy var horizontalStackView: UIStackView = {
+        let stackView = UIStackView()
+        stackView.translatesAutoresizingMaskIntoConstraints = false
+        stackView.axis = .horizontal
+        stackView.alignment = .top
+
+        return stackView
+    }()
+
+    lazy var imageView: UIImageView = {
+        let imageView = UIImageView()
+        imageView.translatesAutoresizingMaskIntoConstraints = false
+        imageView.contentMode = .scaleAspectFill
+        imageView.masksToBounds = true
+        imageView.layer.masksToBounds = true
+        imageView.layer.cornerRadius = 4
+        return imageView
+    }()
+
+
+    lazy var headerTitleView: DiffHeaderTitleView = {
+        let titleView = DiffHeaderTitleView()
+        titleView.translatesAutoresizingMaskIntoConstraints = false
+        titleView.masksToBounds = true
+        return titleView
+    }()
+
+    override func setup() {
+        addSubview(horizontalStackView)
+        horizontalStackView.addArrangedSubview(headerTitleView)
+
+
+            NSLayoutConstraint.activate([
+                horizontalStackView.topAnchor.constraint(equalTo: topAnchor, constant: 10),
+                horizontalStackView.bottomAnchor.constraint(equalTo: bottomAnchor, constant: -10),
+                horizontalStackView.leadingAnchor.constraint(equalTo:  layoutMarginsGuide.leadingAnchor, constant: 10),
+                horizontalStackView.trailingAnchor.constraint(equalTo: layoutMarginsGuide.trailingAnchor, constant: -10)
+
+            ])
+        horizontalStackView.spacing = 16
+        horizontalStackView.addArrangedSubview(imageView)
+
+        NSLayoutConstraint.activate([
+            imageView.heightAnchor.constraint(equalToConstant: 80),
+            imageView.widthAnchor.constraint(equalToConstant: 80)
+        ])
+
+    }
+
+    func configure(with vm: DiffHeaderViewModel) {
+        self.viewModel = vm
+        if let viewModel {
+            updateTitleView(with: viewModel.title)
+            updateImageView(with: viewModel)
+        }
+    }
+
+    func updateImageView(with new: DiffHeaderViewModel) {
+        self.viewModel = new
+
+        if let leadImageURL = viewModel?.imageURL {
+            imageView.wmf_setImage(with: leadImageURL, detectFaces: true, onGPU: true, failure: { (error) in
+                DDLogError("Failure loading diff header image: \(error)")
+            }, success: { [weak self] in
+                self?.imageView.isHidden = false
+            })
+        } else {
+            imageView.isHidden = true
+        }
+    }
+
+    func updateTitleView(with viewModel: DiffHeaderTitleViewModel) {
+        headerTitleView.update(viewModel)
+    }
+
```

Comment:
To support the underbar hit fallthrough to allow a user to scroll the view when interacting with the header itself, we need to add something like this (which we also use in other views): 

```suggestion
override func point(inside point: CGPoint, with event: UIEvent?) -> Bool {
    guard !UIAccessibility.isVoiceOverRunning else {
        return super.point(inside: point, with: event)
    }

    return false
}
```

Reply:
Ohh nice, I didn't know about that 😄 it looks like it's working as expected now!